### ISSUE TITLE
feature request: check doas and run0 similarly to sudo

### ISSUE THREAD

**Issue by e-kwsm** [koalaman/shellcheck#3255]
#### For new checks and feature suggestions
- Rule Id: SC2024, SC2032, SC2033, and SC2232
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

doas and sytemd&rsquo;s run0 are alternatives to sudo, but they are not checked at the moment.

#### Here's a snippet or screenshot that shows a potential problem:

```sh
#!/bin/sh
# SC2024
sudo echo 1 > foo
run0 echo 1 > bar

# SC2032, SC2033
foo() { :; }
sudo foo
doas foo

# SC2232
( sudo cd / )
( run0 cd / )
```

#### Here's what shellcheck currently says:

```
-:3:13: warning: sudo doesn't affect redirects. Use ..| sudo tee file [SC2024]
-:7:1: note: This function can't be invoked via sudo on line 8. [SC2032]
-:8:6: warning: Shell functions can't be passed to external commands. Use separate script or sh -c. [SC2033]
-:12:3: warning: Can't use sudo with builtins like cd. Did you want sudo sh -c .. instead? [SC2232]
```

#### Here's what I wanted to see:

```diff
 -:3:13: warning: sudo doesn't affect redirects. Use ..| sudo tee file [SC2024]
+-:4:13: warning: run0 doesn't affect redirects. Use ..| run0 tee file [SC2024]
 -:7:1: note: This function can't be invoked via sudo on line 8. [SC2032]
+-:7:1: note: This function can't be invoked via doas on line 8. [SC2032]
 -:8:6: warning: Shell functions can't be passed to external commands. Use separate script or sh -c. [SC2033]
+-:9:6: warning: Shell functions can't be passed to external commands. Use separate script or sh -c. [SC2033]
 -:12:3: warning: Can't use sudo with builtins like cd. Did you want sudo sh -c .. instead? [SC2232]
+-:13:3: warning: Can't use run0 with builtins like cd. Did you want run0 sh -c .. instead? [SC2232]
```
